### ISSUE TITLE
Devices are recognized but stated as "No components" [BUG]

### ISSUE THREAD

**Issue by jabss** [yllibed/Zigbee2MqttAssistant#359]
# Describe the bug
Nodes are successfully integrated in zigbee2MQTT. I can see their flows arriving to nodered.
The devices seem to be recognized in zigbee2mqttassistant - Their picture automagically appears in the devices tab.
However, it is not possible to see their components neither configure anything.
The message in the device tab is "**no components** - Maybe this device is not (yet) supported by Zigbee2Mqtt?"
The status tab states the bridge state as "online".
I thought I was something related with the brand of the end devices, but I'm using LIDL, IKEA and SONOFF end devices and the behavior is the same for all of them.

## To Reproduce
Integrate the nodes as normal (press the "allow devices to join network" button on the status tab and run the specific join mode of each end device).
The new devices will appear in the map and in the devices tab, but will have the "no components" message.

Using the latest version of IOTstack.

**Expected behavior**
It should be possible to see which components compose each end-device, specially if they are recognized (their picture "automagically" appears"

![image](https://user-images.githubusercontent.com/7312250/105033962-b970e800-5a50-11eb-8a98-7a6542b3dcf2.png)

**Comment by leiweke** [comment]
I have the same issue
Version of Zigbee2MqttAssistant (this software): 0.3.157 (0.3.157+Branch.master.Sha.ea388c3bbc2d43c4c9c62f27bb6764911ba4acc9) Release
Version of Zigbee2Mqtt: 1.17.0 
Coordinator version: 20190608

**Comment by carldebilly** [comment]
@leiweke @jabss 
It's happening when Z2MA is now receiving the "discovery" topics. If you restart Z2MA, is it solving the problem?

**Comment by jabss** [comment]
I restarted the zigbee2mqtt container several times, and still experience the same issue.
My versions:

`Version of Zigbee2MqttAssistant (this software): 0.3.157 (0.3.157+Branch.master.Sha.ea388c3bbc2d43c4c9c62f27bb6764911ba4acc9) Release
Version of Zigbee2Mqtt: 1.16.2
Coordinator version: 20201026
Coordinator Zigbee Id: 0x00124b0021cc3d98`

**Comment by leiweke** [comment]
same here restarted several time, still the same issue:

Version of Zigbee2MqttAssistant (this software): 0.3.157 (0.3.157+Branch.master.Sha.ea388c3bbc2d43c4c9c62f27bb6764911ba4acc9) Release
Version of Zigbee2Mqtt: 1.17.0 
Coordinator version: 20190608 
Coordinator Zigbee Id: 0x00124b0014da3999

**Comment by suprimex** [comment]
exactly same problem, latest versions.
![image](https://user-images.githubusercontent.com/4375164/105874629-78a34100-6005-11eb-993a-f9969e6ef969.png)

**Comment by Neumi12** [comment]
as in #355 suggested i tried to enable homeassistent and it worked!

**Comment by jabss** [comment]
> 
> 
> as in #355 suggested i tried to enable homeassistent in the config and it worked!

I don't have homeassistant. Is there any implication of turning this variable to true?

**Comment by Neumi12** [comment]
I dont't have homeassistant either, but it solved the issue for me.

**Comment by jabss** [comment]
ok, for future reference:

I've changed the file  ` /home/pi/IOTstack/volumes/zigbee2mqtt/data/configuration.yaml`  in the first line by changing
` homeassistant: false`  to ` homeassistant: true`.

Adding ` homeassistant: true` in the `/home/pi/IOTstack/docker-compose.yml` file didn't seem to work:

```
pi@raspberrypi:~/IOTstack $ docker-compose up -d
ERROR: The Compose file './docker-compose.yml' is invalid because:
Unsupported config option for services.zigbee2mqtt: 'homeassistant'
```
I wonder how to make this configuration persistent. I guess if I update zigbee2MQTT this value will be reset.

But yeah, now I can see the device components in the devices tab.
It seems it was a configuration issue on the zigbee2MQTT rather than a bug in the zigbee2MQTTAssistant. Will close the issue and hope I've documented clearly how to overcome the situation.

**Comment by jabss** [comment]
Not a bug in zigbee2MQTTAsssitant.
It is a configuration in the zigbee2MQTT.
Closing.

**Comment by leiweke** [comment]
maybe that is not correct, I have no HA and if I change to homeassistant: true  Z2M will not start!! change to homeassistant: false it starts correctly. so that is not the fix ! It must be Z2M Assistant.

**Comment by jabss** [comment]
I have rebooted the whole system (RPI4 with IoTStack that has the zigbee2MQTT et all) with the mentioned change and I'm still able to receive warning in my smartwatch when the house doors are opened (Sensor device --> zigbee2MQTT --> NodeRed --> Telegram --> Internet --> My smartphone --> My smartwatch).
Therefore I can say it is working in my case and I also don't have Homeassistant. 
Where did you change the parameter? I changed it in /home/pi/IOTstack/volumes/zigbee2mqtt/data/configuration.yaml.
It seems ZB2MQTT only passes that component info to ZB2MQTTAssistant if that flag is true, so it doesn't seem to be a bug in ZB2MQTTAssistant.
